src/T2DMap.cpp

```diff
@@ -900,6 +909,15 @@ inline void T2DMap::drawRoom(QPainter& painter, QFont& roomVNumFont, QPen& pen,
         }
 
     }
+    // Do we need to draw the room name:
+    if (mShowRoomName && areRoomIdsLegible) {
+        painter.save();
+        QColor roomIdColor;
+        painter.setPen(QPen(QColor(Qt::white)));
```

Comment:
What about if the map background is not a dark colour - you will want to consider the map background colour (The `(QColor) mpHost->mBgColor_2` variable accessible in the `(void) T2DMap::paintEvent(QPaintEvent*)` method and use `Qt::white` if the `QColor::lightness()` is less than 128 or `Qt::black` if it above.

Reply:
will fix